ridley/src/System/Metrics/Prometheus/Ridley.hs

```diff
@@ -201,5 +205,16 @@ serveMetrics = P.serveHttpTextMetrics
 #endif
 
 --------------------------------------------------------------------------------
-updateHandlers :: [RidleyMetricHandler] -> IO ()
-updateHandlers = mapM_ runHandler
+updateHandlers :: LogEnv -> [RidleyMetricHandler] -> IO ()
+updateHandlers le hs = mapM_ (\h@RidleyMetricHandler{..} -> runHandler h `Ex.catchAny` (logFailedUpdate le label _cs)) hs
+
+logFailedUpdate :: LogEnv -> T.Text -> CallStack -> Ex.SomeException -> IO ()
+logFailedUpdate le lbl cs ex =
+  runKatipContextT le () "errors" $ do
+      $(logTM) ErrorS $
+        fromString $ "Couldn't update handler for "
+                  <> "\"" <> T.unpack lbl <> "\""
+                  <> " originally defined at "
+                  <> prettyCallStack cs
+                  <> " due to "
+                  <> Ex.displayException ex
```

Comment:
I (still) think the message should have the exception before the call stack (because the latter breaks over multiple lines). Otherwise this looks reasonable, though I'm not familiar with the guts of `ridley`.

Reply:
Ack, I will swap the order and merge this one. Thanks! 😸